pkgs/development/libraries/science/math/magma/generic.nix

```diff
@@ -119,14 +126,16 @@ stdenv.mkDerivation {
     cmake
     ninja
     gfortran
+  ] ++ lists.optionals cudaSupport [
+    cuda-native-redist
   ];
 
   buildInputs = [
     libpthreadstubs
     lapack
     blas
   ] ++ lists.optionals cudaSupport [
-    cuda_joined
+    cuda-redist
```

Comment:
Do we drop support for `cudaPackages_10`?

Reply:
Hmm... I realize that because I'm using `cudaPackages`, this won't work with anything prior to CUDA 11.4, since that's when we got redistributables, right? In that case, I guess I accidentally dropped support in https://github.com/NixOS/nixpkgs/pull/217410.

Thoughts? Magma can (does?) support CUDA 10, but the only consumer in Nixpkgs is PyTorch, and they dropped support for CUDA 10 in their latest release.

Reply:
> won't work with anything prior to CUDA 11.4

Yes

> dropped support in ...

Oh, indeed. Maybe it's just time to prune and drop a notice in release notes. We can keep `cudaPackages_10` around for legacy packages that won't work without it, but when a package supports redist, let's default to only supporting redist. Thoughts? CC @samuela 

<details>
at least right now my feeling is that it's better to support well the things we actually use, than do a poor job at stretching for things we don't even know users for
</details>

Reply:
> Oh, indeed. Maybe it's just time to prune and drop a notice in release notes. We can keep cudaPackages_10 around for legacy packages that won't work without it, but when a package supports redist, let's default to only supporting redist. Thoughts? CC @samuela

Agreed, we can't let these outdated versions hold us back from progress!